metricbeat/module/kubernetes/util/metrics_cache.go

```diff
@@ -68,6 +59,41 @@ func (c *PerfMetricsCache) Stop() {
 	c.ContainerCoresLimit.Stop()
 }
 
+func (c *PerfMetricsCache) GetTimeout() time.Duration {
+	var ans time.Duration = 0
+
+	nmATimeout := c.NodeMemAllocatable.GetTimeout()
+	if nmATimeout > ans {
+		ans = nmATimeout
+	}
+
+	ncATimeout := c.NodeCoresAllocatable.GetTimeout()
+	if ncATimeout > ans {
+		ans = ncATimeout
+	}
+
+	cmLTimeout := c.ContainerMemLimit.GetTimeout()
+	if cmLTimeout > ans {
+		ans = cmLTimeout
+	}
+
+	ccLTimeout := c.ContainerCoresLimit.GetTimeout()
+	if ccLTimeout > ans {
+		ans = ccLTimeout
+	}
+	return ans
+}
+
+func (c *PerfMetricsCache) SetTimeout(timeout time.Duration) {
```

Comment:
nit: 
```suggestion
func (c *PerfMetricsCache) SetOrUpdateTimeout(timeout time.Duration) {
```

Also could you please add a docstring for this method?